src/hooks/useUsers.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from "react";
+import { useFirestoreConnect } from "react-redux-firebase";
+
+import { usersSelector, usersByIdSelector } from "utils/selectors";
+import { WithId } from "utils/id";
+
+import { useSelector } from "hooks/useSelector";
+import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
+import { useVenueId } from "hooks/useVenueId";
+
+import { User } from "types/User";
+
+const useConnectUsers = () => {
+  useFirestoreConnect("users");
+};
+
+export const usePartygoers = () => {
+  useConnectUsers();
+
+  const lastSeenThreshold = useUserLastSeenThreshold();
+
+  const users = useSelector(usersSelector);
+
+  return useMemo(
+    () => users.filter((user) => user.lastSeenAt > lastSeenThreshold),
+    [users, lastSeenThreshold]
+  );
+};
+
+export const useUsersById = () => {
+  useConnectUsers();
+
+  return useSelector(usersByIdSelector);
+};
+
+export const useVenueUsers = () => {
+  const venueId = useVenueId();
+  const users = useSelector(usersSelector);
+
+  return useMemo(
+    () =>
+      users.filter(
+        (user) => venueId && user.enteredVenueIds?.includes(venueId)
+      ),
+    [users, venueId]
+  );
+};
+
+export const useCampPartygoers = (venueName: string): WithId<User>[] => {
+  const partygoers = usePartygoers();
+
+  const lastSeenThreshold = useUserLastSeenThreshold();
+
+  return useMemo(
+    () =>
+      partygoers.filter(
+        (partygoer) => partygoer?.lastSeenIn?.[venueName] > lastSeenThreshold
+      ),
+    [partygoers, lastSeenThreshold, venueName]
+  );
+};
```

Comment:
Okay, I think I will have to delete this hook when you remove Camp and Playa from the codebase. We still have usePartygoers hook used across the codebase

Reply:
For my knowledge, what (if any) changes did you make in e7c05d2e2c040fd58dcda1e648598489d528fd44 related to this comment? Hard to figure out what was done in relation to which comments, etc.

Reply:
Didn't do any changes, since this hook is still used in Camp. When you merge the branch with deletion of that, I will create a PR to remove what is left unused in the hooks

Reply:
Ok, makes sense. It may even be that I clean it up in my camp PR, as i'm going to need to update it against staging (and I assume this PR will be merged by then)